lib/Language/Souffle/Interpreted.hs

```diff
@@ -287,11 +315,25 @@ readCSVFile path = doesFileExist path >>= \case
 --   This functionality is only provided for the interpreted version since the
 --   compiled version directly (de-)serializes data via the C++ API.
 cleanup :: forall prog. Program prog => Handle prog -> SouffleM ()
-cleanup (Handle ref) = liftIO $ do
-  handle <- readIORef ref
+cleanup (Handle (refHandleData,refHandleStdOut,refHandleStdErr)) = liftIO $ do
+  handle <- readIORef refHandleData
   traverse_ removeDirectoryRecursive [factPath handle, outputPath handle, basePath handle]
+  readIORef refHandleStdOut >>= mapM_ hClose
+  readIORef refHandleStdErr >>= mapM_ hClose
```

Comment:
Would `fmap hClose <$> readIORef refHandleStdOut` (and same for stderr) work here also?